src/components/ChatPanel.tsx

```diff
@@ -157,25 +161,40 @@ export function ChatPanel(props: ChatPanelProps) {
   const messagesRef = useRef<Array<HTMLDivElement | null>>([]);
   const messagesContainer = useRef<HTMLDivElement>(null);
 
+  // State to help detect initial messages rendering
+  const [initialMessagesLength] = useState(messages.length);
+
+  const savedPanelState = useMemo(() => {
+    if (!conversationId || !messages.length) {
+      return {};
+    }
+    return loadSessionState(conversationId);
+  }, [conversationId, messages]);
```

Comment:
this will attempt to load session on every new message. I think it's fine to just check for conversationId?